CONTRIBUTING.md

```diff
@@ -1 +1,42 @@
-# Contributing
+# Contributing to mgcpy
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+
+## We develop with GitHub
+We use GitHub to host code, to track issues and feature requests, as well as accept pull requests.
+
+## Use a consistent coding style
+We use [pep8 style guide](https://www.python.org/dev/peps/pep-0008/).
+To setup,
+- `pip install pycodestyle`
+- Run `./install_hooks.sh` to install the pre-commit hooks for enforcing **pep8** before committing any code.
+
+## Report bugs (using GitHub's issues)
+We use GitHub issues to track public bugs. Report a bug by [opening a new issue](https://github.com/NeuroDataDesign/mgcpy/issues), it's that easy!
+- Write bug reports with detail, background, and sample code
+- **Great Bug Reports** tend to have:
+
+  - A quick summary and/or background
+  - Steps to reproduce
+    - Be specific!
+    - Give sample code
+  - What you expected would happen
+  - What actually happens
+  - Notes (possibly including why you think this might be happening, or stuff you tried that didn't work, screenshots, etc.)
+
+## Proposing new features / fixes
+Pull requests are the best way to propose changes to the codebase (we use [GitHub Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
```

Comment:
do we have github flow setup?

Reply:
I think it just a guideline to follow and not something to setup.

Reply:
Oh i see